ironfish/src/account/accounts.ts

```diff
@@ -616,12 +616,18 @@ export class Accounts {
   ): Promise<ReadonlyArray<{ hash: string; note: Note; index: number | null }>> {
     const unspentNotes = []
 
+    const notePromises = []
     for (const transactionMapValue of this.transactionMap.values()) {
-      const result = await this.workerPool.getUnspentNotes(
-        transactionMapValue.transaction.serialize(),
-        [account.incomingViewKey],
+      notePromises.push(
+        this.workerPool.getUnspentNotes(transactionMapValue.transaction.serialize(), [
+          account.incomingViewKey,
+        ]),
       )
+    }
+
+    const results = await Promise.all(notePromises)
 
+    for (const result of results) {
```

Comment:
Starving the worker pool is potentially an issue here, although these jobs are fairly fast. If you'd feel more comfortable, we could do this in batches: x transactions at a time, await, process, x more, etc. For what it's worth, running 2 accounts:balance on a loop on my test chain isn't enough to back up my worker pool, although it does kick off ~2k jobs.

A potential medium-term fix could be to create a job to batch these so you're kicking off less overall jobs, spending less overhead on creating jobs, etc. I'm not sure if that's worth the effort given we're on verge of beginning the bigger accounts work.

I do agree that this isn't the long-term fix, for sure. The goal here is just a quick-win, whereas the better, bigger fix will likely come with the accounts stuff in the following weeks,

Reply:
On the topic of worker pool starvation (not the actual logic change Jason mentioned):

Does our workerPool not have a queueing system so you can call workerPool as many times as needed and it will just adjust to the resources it has? Seems bad that a client of the workerPool would have to think about starvation at all.

Reply:
My reading of the code is that we are still doing the same amount of worker tasks though? We are just artificially rate limiting by awaiting the result of each.

The idea that you have to call await to "protect" the workerPool seems pretty fundamentally misaligned with how I would expect a workerPool to function. We should just dump work on it, and if it can't handle the work we optimize how the pool works instead of having the workerPool client code figure out ways to gatekeep/rate limit.

Generally, I'm a bit surprised at the idea that workerPool usage feels like a bottleneck. I would expect us to have to be pretty finely optimized before running a task on a thread becomes a material amount of overhead.